sql/head.sql

```diff
@@ -0,0 +1,40 @@
+
+set local search_path = pg_catalog, pg_temp;
+
+/*
+if this is the first installation of the extension, create the migration table.
+if the extension has already been installed (i.e. we are upgrading), make sure
+that the user doing the upgrade is the same user that owns the ai_migration
+table. abort the upgrade if different.
```

Comment:
Is this a common practice? Can't we rely on regular table permissions?

Reply:
We had this check in Promscale. I know it was security-related. We don't want one database user to be able to take over the extension from another. I don't remember the specifics of the vulnerability. Maybe @cevian does?

Reply:
Okay, I remember the vulnerability. If a user "bob" creates the schema "ai" and the migration table "ai.migration" and THEN someone does `create extension ai cascade;` then "bob" can do nasty, hateful things.